src/test/java/com/google/devtools/build/lib/bazel/rules/genrule/GenRuleWindowsTest.java

```diff
@@ -0,0 +1,142 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.bazel.rules.genrule;
+
+import static com.google.common.truth.Truth.assertThat;
+import com.google.devtools.build.lib.actions.Artifact;
+import com.google.devtools.build.lib.analysis.actions.SpawnAction;
+import com.google.devtools.build.lib.analysis.util.BuildViewTestCase;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests of {@link BazelGenRule} on Windows.*/
+@RunWith(JUnit4.class)
+public class GenRuleWindowsTest extends BuildViewTestCase {
+
+  private static final Pattern BATCH_COMMAND_PATTERN =
+      Pattern.compile("(?<command>.*)");
```

Comment:
Hm, isn't this regex equivalent to `.*`, i.e. matching any string?

Reply:
Ah, that's true. Let me remove the redundancy.